src/goto-programs/rw_set.cpp

```diff
@@ -49,6 +49,37 @@ void rw_sett::read_write_rec(
       {
         return; // ignore for now
       }
+
+      //improvements for CUDA features
+      if(
+        expr.type().tag() == "__dim3" ||
+        expr.type().tag().as_string().find("arg_struct") != std::string::npos ||
+        symbol->name.as_string().find("__dim3") != std::string::npos ||
+        symbol->name.as_string().find("io") != std::string::npos ||
+        symbol->name.as_string().find("fopen") != std::string::npos ||
+        symbol->name.as_string().find("pthread_lib") != std::string::npos ||
+        symbol->name.as_string().find("__CPROVER") != std::string::npos ||
+        symbol->name.as_string().find("__cuda") != std::string::npos ||
+        symbol->name.as_string().find("__ESBMC") != std::string::npos ||
+        symbol->name.as_string().find("cuda") != std::string::npos ||
+        symbol->name.as_string().find("itoa") != std::string::npos ||
+        symbol->name.as_string().find("mem") != std::string::npos ||
+        symbol->name.as_string().find("rev") != std::string::npos ||
+        symbol->name.as_string().find("search") != std::string::npos ||
+        symbol->name.as_string().find("std") != std::string::npos ||
+        symbol->name.as_string().find("str") != std::string::npos ||
+        symbol->name.as_string().find("verify_kernel") != std::string::npos ||
+        symbol->name.as_string().find("virtual_table") != std::string::npos ||
```

Comment:
Neither of these `.find()` conditions is a good idea: for instance, a substring like `str` or `std` or `mem` is way too common in regular code. If there is any way to use the `__ESBMC` **prefix** that would help in making these exceptions to read_write_rec() apply only to known symbols instead of matching arbitrary unknown ones.

If that's not possible, could you explain why? In any case we should find a way to make it possible or otherwise disambiguate the symbol names.